core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,18 +145,54 @@ oppia.factory('SimpleEditorManagerService', [
           data.questionList.isEmpty() ?
           SimpleEditorShimService.getInitStateName() :
           data.questionList.getLastQuestion().getDestinationStateName());
-
         SimpleEditorShimService.saveInteractionId(
-          lastStateName, DEFAULT_INTERACTION.ID);
+          lastStateName, 'MultipleChoiceInput');
         SimpleEditorShimService.saveCustomizationArgs(
-          lastStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+          lastStateName,
+          DEFAULT_INTERACTION_PROPERTIES.MultipleChoiceInput.CUSTOMIZATION_ARGS
+        );
         SimpleEditorShimService.saveDefaultOutcome(
           lastStateName, OutcomeObjectFactory.createEmpty(lastStateName));
 
         var stateData = SimpleEditorShimService.getState(lastStateName);
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(newQuestionType, index) {
+        var currentStateName = data.questionList.getAllStateNames()[index];
+        var nextStateName = data.questionList.getAllStateNames()[index + 1];
+        var questionCount = data.questionList.getQuestionCount();
+        var currentInteractionId = (
+          SimpleEditorShimService.getInteractionId(currentStateName));
+        var doesLastQuestionHaveAnswerGroups = (
+          QuestionListObjectFactory.doesLastQuestionHaveAnswerGroups);
+
+        // Update Question Type If interactionId is not same.
+        if (newQuestionType !== currentInteractionId) {
+          var newAnswerGroups = [];
+          SimpleEditorShimService.saveInteractionId(
+            currentStateName, newQuestionType);
+          SimpleEditorShimService.saveCustomizationArgs(
+            currentStateName,
+            DEFAULT_INTERACTION_PROPERTIES[newQuestionType].CUSTOMIZATION_ARGS);
+          newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+            RuleObjectFactory.createNew(
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.type,
+              DEFAULT_INTERACTION_PROPERTIES[newQuestionType].
+                FIRST_ANSWER_GROUP_RULE.value
+            )
+          ], OutcomeObjectFactory.createEmpty(nextStateName), false));
+
+          if (doesLastQuestionHaveAnswerGroups && index !== questionCount - 1) {
```

Comment:
I think I'm missing something here. If you have questions 1, 2, 3, 4, 5 and you change the question type of question 2, then what does the last question's answer groups have to do with it? Also, if there's a scenario where you aren't going to save the new answer groups, then why create them at all in that case (i.e. why doesn't the update of newAnswerGroups go into the 'if' statement)?